zerver/migrations/0479_realm_backfill_uuid_and_secret.py

```diff
@@ -0,0 +1,67 @@
+import secrets
+import uuid
+
+from django.db import migrations, models
+from django.db.backends.base.schema import BaseDatabaseSchemaEditor
+from django.db.migrations.state import StateApps
+
+
+def generate_api_key() -> str:
+    """
+    This is a copy of zerver.lib.utils.generate_api_key
+    """
+
+    api_key = ""
+    while len(api_key) < 32:
+        api_key += secrets.token_urlsafe(3 * 9).replace("_", "").replace("-", "")
+    return api_key[:32]
+
+
+def generate_realm_uuid_owner_secret() -> str:
+    token = generate_api_key()
+
+    return f"zulipapi_{token}"
+
+
+def backfill_realm_uuid_and_secret(
+    apps: StateApps, schema_editor: BaseDatabaseSchemaEditor
+) -> None:
+    Realm = apps.get_model("zerver", "Realm")
+
+    max_id = Realm.objects.aggregate(models.Max("id"))["id__max"]
+    if max_id is None:
+        # Nothing to do if there are no realms yet.
+        return
+
+    BATCH_SIZE = 10000
```

Comment:
1000 is probably a better batch size for secret generation -- it can be a somewhat expensive process.

Reply:
Though actually we might want more like 100 given that `bulk_update` is accidentally quadratic in how it generates queries, right?

Reply:
Makes sense, changed it to 100